src/AutoRest.CSharp/Output/Models/Types/ObjectType.cs

```diff
@@ -111,6 +114,12 @@ public ObjectType(ObjectSchema objectSchema, BuildContext context, bool isResour
         public ObjectTypeConstructor InitializationConstructor => _initializationConstructor ??= BuildInitializationConstructor();
         public ObjectTypeConstructor SerializationConstructor => _serializationConstructor ??= BuildSerializationConstructor();
 
+        public void OverrideInherits(CSharpType cSharpType)
```

Comment:
Please don't make models returned from the OutputLibrary mutable. They interact in many ways and values can already be observed before we override them.

Reply:
Added ADO item to track this after the merge into feature/v3
https://dev.azure.com/azure-mgmt-ex/DotNET%20Management%20SDK/_workitems/edit/5724